core/network/impl/router_libp2p.cpp

```diff
@@ -96,37 +96,36 @@ namespace kagome::network {
   }
 
   bool RouterLibp2p::start() {
-    // IPv6
-    {
-      auto ma_res = libp2p::multi::Multiaddress::create(
-          "/ip6/::/tcp/" + std::to_string(app_config_.p2pPort()) + "/p2p/"
-          + own_info_.id.toBase58());
-      BOOST_ASSERT(ma_res.has_value());
-      auto &ma = ma_res.value();
-      auto res = host_.listen(ma);
-      if (not res) {
-        log_->error("Cannot listen address {}. Error: {}",
-                    ma.getStringAddress(),
-                    res.error().message());
+    auto listen_addresses = app_config_.listenAddresses();
+
+    using P = libp2p::multi::Protocol::Code;
+    for (auto &listen_address : listen_addresses) {
+      if (listen_address.getProtocols().size()
+              < 3  // does not have /p2p or alternative part
+          and (listen_address.hasProtocol(P::IP4)
```

Comment:
Do we need to process /dns4 /dns6 /dns /dnsaddr family of addresses?

Reply:
Discussed offline. Here we deal with interface addresses, thus only IP4 and IP6 addresses support is required at this place.